src/Symfony/Component/Messenger/Tests/WorkerTest.php

```diff
@@ -245,6 +247,37 @@ public function testWorkerWithMultipleReceivers()
         $this->assertSame([$envelope1, $envelope2, $envelope3, $envelope4, $envelope5, $envelope6], $processedEnvelopes);
     }
 
+    public function testWorkerLimitQueues()
+    {
+        $receiver1 = $this->createMock(QueueReceiverInterface::class);
+        $receiver1->expects($this->once())
+            ->method('getFromQueues')
+            ->with(['foo'])
+            ->willReturn([new Envelope(new DummyMessage('message1'))])
+        ;
+
+        $bus = $this->getMockBuilder(MessageBusInterface::class)->getMock();
+
+        $dispatcher = new EventDispatcher();
+        $dispatcher->addSubscriber(new StopWorkerOnMessageLimitListener(1));
+
+        $worker = new Worker([$receiver1], $bus, $dispatcher);
+        $this->expectException(RuntimeException::class);
+        $worker->run(['queues' => ['foo']]);
+    }
+
+    public function testWorkerLimitQueuesUnsupported()
+    {
+        $receiver1 = $this->createMock(QueueReceiverInterface::class);
+        $receiver2 = $this->createMock(ReceiverInterface::class);
+
+        $bus = $this->getMockBuilder(MessageBusInterface::class)->getMock();
+
+        $worker = new Worker([$receiver1, $receiver2], $bus);
+        $this->expectException(RuntimeException::class);
```

Comment:
Since this is not a specific exception class, could we assert parts of the exception message?

Reply:
fixed